### ISSUE TITLE
User permissions are too open

### ISSUE THREAD

**Issue by J-ongres** [supabase-community/supabase-kubernetes#37]
# Bug report

## Describe the bug

The permissions are too open. The `supabase_admin` user is a superuser and can do pretty much everything to the database.

When we're using an approach where the schema can be initialized by a superuser, there should be no need to have more permissions after the schema has been created (here by a script).

This should be restricted to an exhaustive list of permissions and what's required on Supabase's side. It should also be defined more clearly which users/roles are required from Supabase's side during runtime.

**Comment by luizfelmach** [comment]
Thanks for raising this — the concern around permissions and long-lived superuser roles is absolutely valid, especially from a security and hardening perspective.

That said, this repository mainly acts as a Kubernetes-focused mirror / adaptation of the official Supabase setup from github.com/supabase/supabase, rather than defining or owning Supabase’s core database security model.

Because of that, questions around which roles are required at runtime, how permissions should be restricted after bootstrap, and what the long-term permission model should look like are likely better addressed in the main Supabase repository.

If you still consider these permission concerns relevant (which is very reasonable), I’d strongly suggest opening or linking an issue in github.com/supabase/supabase, where the core database roles and security assumptions are defined and maintained.

That discussion would also help clarify what guarantees and threat model Supabase officially supports, which would then naturally flow back into Kubernetes-based deployments like this one.

**Comment by aantti** [comment]
Correct, thanks for the explanation above, @luizfelmach. In turn, the "upstream" Docker Compose based [configuration](https://supabase.com/docs/guides/self-hosting/docker) for self-hosted largely matches the Supabase platform projects configurations. That said, migrating from `supabase_admin` to `postgres` for both [self-hosted](https://supabase.com/docs/guides/self-hosting) and [CLI/local-dev](https://supabase.com/docs/reference/cli/introduction) is planned. It is a valid concern that `supabase_admin` is overprivileged when used with Studio, etc.

**Comment by luizfelmach** [comment]
Thanks everyone for the discussion and clarifications.

Since this repository follows the upstream Supabase configuration and there’s already alignment (including the planned migration away from supabase_admin), I’m going to close this PR.

Further changes to the database permission model should be discussed and implemented upstream in the official Supabase repositories, and then reflected here if needed.

Appreciate the feedback and security concerns raised here.
